core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -333,22 +333,25 @@ object Extraction {
 
     def newTuple(root: JValue, mappings: List[Mapping]): Any = {
       root match {
-        case JArray(items) if items.length >= 1 && items.length <= 22 =>
+        case JArray(items) if items.nonEmpty && items.length <= tuples.length =>
           val builtItems: Seq[Object] = items.zip(mappings).map({
-            case (i,m) =>
-              build(i, m).asInstanceOf[Object]
+            case (item, mapping) =>
+              build(item, mapping).asInstanceOf[Object]
           })
-          val tupleClass = Class.forName("scala.Tuple" + items.length)
+          val tupleIndex = items.length - 1
+          val tupleClass = tuples.lift(tupleIndex).getOrElse {
+            throw new IllegalArgumentException(s"Cannot instantiate a tuple of length ${items.length} even though that should be a valid tuple length.")
```

Comment:
Sorry, basically doing something like this:

```scala
val tupleClasses =
  Seq(classOf[Tuple1[_]]...)
val tupleConstructors =
  tupleClasses.map(clazz => (clazz -> clazz.getConstructors()(0))).toMap
```

And then looking the constructor up in the map every time.